### ISSUE TITLE
[BUG] Recursively stacked parts will cease to respect linkages after one or two parts stacked

### ISSUE THREAD

**Issue by boot2big** [DonBruce64/MinecraftTransportSimulator#1548]
Polar opposite of #1521
Here's an example illustration to describe where stacked parts cease to be linked to engines, controller seats, etc.

1. Vehicle, 2. Hydraulics Kit, 3. Hover Kit, 4. Craftspeed Compat Part, 5. Actual wheel

1 is the vehicle itself, 5 is what is supposed to ultimately be driven. However, 1 will only provide linked power to 3, and anything past 3 (starting at 4, and 5 is right out) will no longer receive power. 

To put it simply, using the above diagram would result in a very capable and slick but sleeper setup on a vehicle... that won't be able to move. Even though by all accounts, 5 is on 4, which is on 3, which is still getting linked power. And if you activate the hover kit, your vehicle will be able to move based on whatever that hover wheel is doing.

**Comment by DonBruce64** [comment]
So the reason this doesn't work is because the system only checks for linking down the tree until it finds something.  It'll go and look at all parts, and when it finds a wheel, it'll stop there, and won't check that wheel for sub-parts since it assumes there won't be any wheels on wheels, or guns on guns, that should be linked as such.  Is this an incorrect assumption?  Should everything in the slot be fair-game, including all parts on the linked part in the slot, irregardless of them being on an already-linked part?

**Comment by boot2big** [comment]
With you pointing out that this is intentional, I suppose I'd have to agree that this is a correct assumption then, as there's no other pack that adds as many not-wheel wheel parts as my pack, and the only ones that may ever do so will stick to having specifically-defined wheel slots (such as UNU having tires that can be installed onto the wheels themselves) that can't be stacked.

I still would appreciate a change if you're ever in the mood to do so, but it is not urgent and can be backburnered as an enhancement to focus on proper bugs and 1.16 development. The only feature that is ultimately hindered by this is something that I doubt most people will do anyways as it isn't common for anyone to make a donked-out lowrider hover car.

**Comment by DonBruce64** [comment]
I'm not saying it can't be changed.  It very well can.  More *should* it be changed?  Are there any packs that will break if one links things that far deep?  Wheels I'm less concerned about (as if you put a wheel part on a wheel, you'll get nasty physics already), but guns are more what I'm concerned about.  In particular mounted units with seats built in to them.  I do think it's not a bad idea to link all possible parts in a linked slot.  But I'm asking my engineers if I should, rather than if I could.

**Comment by boot2big** [comment]
I am unaware of any packs that would be _negatively_ affected by such a change save for possibly IAV's military pack which may have installable sub-machine guns for their tank turrets. I think you would have to check with Furanzu on that, however I can't think of any other packs that would wig out if parts linked more recursively than they do now.

**Comment by DonBruce64** [comment]
@Ilya-torshov

**Comment by boot2big** [comment]
Fixed/implemented in latest 22.6.0-beta (As of 6:47pm)

https://user-images.githubusercontent.com/53581325/222857869-a2b77247-2e03-493a-a39d-b330a3fd0863.mp4

**Comment by DonBruce64** [comment]
Fixed in V22.7.0.
